nullaway/src/test/java/com/uber/nullaway/NullAwayFrameworkTests.java

```diff
@@ -365,4 +365,75 @@ public void systemConsoleNullable() {
+  @Test
+  public void mapGetOrDefault() {
+    makeTestHelperWithArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber"))
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import java.util.Map;",
+            "import com.google.common.collect.ImmutableMap;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "   void testGetOrDefaultMap(Map<String, String> m, String string, @Nullable String nullableString) {",
+            "     m.getOrDefault(\"key\", \"value\").toString();",
+            "     m.getOrDefault(\"key\", string).toString();",
+            "     // BUG: Diagnostic contains: dereferenced",
+            "     m.getOrDefault(\"key\", null).toString();",
+            "     // BUG: Diagnostic contains: dereferenced",
+            "     m.getOrDefault(\"key\", nullableString).toString();",
+            "   }",
+            "",
+            "   void testGetOrDefaultImmutableMap(ImmutableMap<String, String> im, String string, @Nullable String nullableString) {",
+            "     im.getOrDefault(\"key\", \"value\").toString();",
+            "     im.getOrDefault(\"key\", string).toString();",
+            "     // BUG: Diagnostic contains: dereferenced",
+            "     im.getOrDefault(\"key\", null).toString();",
+            "     // BUG: Diagnostic contains: dereferenced",
+            "     im.getOrDefault(\"key\", nullableString).toString();",
+            "   }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void mapGetOrDefaultWithAcknowledgeRestrictiveAnnotations() {
+    makeTestHelperWithArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber",
+                "-XepOpt:NullAway:AcknowledgeRestrictiveAnnotations=true"))
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import java.util.Map;",
+            "import com.google.common.collect.ImmutableMap;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "   void testGetOrDefaultMap(Map<String, String> m, String string, @Nullable String nullableString) {",
+            "     m.getOrDefault(\"key\", \"value\").toString();",
+            "     m.getOrDefault(\"key\", string).toString();",
+            "     // BUG: Diagnostic contains: dereferenced",
+            "     m.getOrDefault(\"key\", null).toString();",
+            "     // BUG: Diagnostic contains: dereferenced",
+            "     m.getOrDefault(\"key\", nullableString).toString();",
+            "   }",
+            "",
+            "   void testGetOrDefaultImmutableMap(ImmutableMap<String, String> im, String string, @Nullable String nullableString) {",
```

Comment:
nit: when i originally copied the example from somewhere there was 3 space indentation on the 1st level, but maybe we want to use 2 instead everywhere

Reply:
Thanks!  Fixed in 979373354d1be9fdb36c2d782f858ac12e1c15b7